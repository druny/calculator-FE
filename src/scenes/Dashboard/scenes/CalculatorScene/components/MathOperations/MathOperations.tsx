import React, { Dispatch } from 'react';
import {
  Add as AddIcon,
  Remove as MinusIcon,
  AcUnitSharp as AcUnitSharpIcon,
  BlurOffSharp as BlurOffSharpIcon,
  SvgIconComponent,
} from '@material-ui/icons';

import { PLUS, MINUS, MULTIPLY, DIVIDE } from '@/constants';

import { OperationButton } from '../OperationButton';

type OperationType = {
  operationName: string;
  operationLogo: SvgIconComponent;
};

type MathOperationsProps = {
  setActiveOperation: Dispatch<string>;
};

const operations: OperationType[] = [
  {
    operationName: PLUS,
    operationLogo: AddIcon,
  },
  {
    operationName: MINUS,
    operationLogo: MinusIcon,
  },
  {
    operationName: MULTIPLY,
    operationLogo: AcUnitSharpIcon,
  },
  {
    operationName: DIVIDE,
    operationLogo: BlurOffSharpIcon,
  },
];

export function MathOperations({ setActiveOperation }: MathOperationsProps) {
  return (
    <div>
      {operations.map(({ operationName, operationLogo }) => (
        <OperationButton
          key={operationName}
          operationLogo={operationLogo}
          onClick={() => setActiveOperation(operationName)}
        />
      ))}
    </div>
  );
}
