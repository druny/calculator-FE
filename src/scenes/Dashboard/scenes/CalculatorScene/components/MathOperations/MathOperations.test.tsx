import React from 'react';
import renderer from 'react-test-renderer';

import { MathOperations } from './MathOperations';

it('renders correctly when there are no items', () => {
  const tree = renderer.create(<MathOperations setActiveOperation={jest.fn()} />).toJSON();
  expect(tree).toMatchSnapshot();
});
