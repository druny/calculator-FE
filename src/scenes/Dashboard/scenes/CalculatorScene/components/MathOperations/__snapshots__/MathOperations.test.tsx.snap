// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13H5v-2h14v2z"
        />
      </svg>
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M22 11h-4.17l3.24-3.24-1.41-1.42L15 11h-2V9l4.66-4.66-1.42-1.41L13 6.17V2h-2v4.17L7.76 2.93 6.34 4.34 11 9v2H9L4.34 6.34 2.93 7.76 6.17 11H2v2h4.17l-3.24 3.24 1.41 1.42L9 13h2v2l-4.66 4.66 1.42 1.41L11 17.83V22h2v-4.17l3.24 3.24 1.42-1.41L13 15v-2h2l4.66 4.66 1.41-1.42L17.83 13H22v-2z"
        />
      </svg>
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <circle
          cx="14"
          cy="6"
          r="1"
        />
        <path
          d="M13.8 11.48l.2.02c.83 0 1.5-.67 1.5-1.5s-.67-1.5-1.5-1.5-1.5.67-1.5 1.5l.02.2c.09.67.61 1.19 1.28 1.28zM14 3.5c.28 0 .5-.22.5-.5s-.22-.5-.5-.5-.5.22-.5.5.22.5.5.5zm-4 0c.28 0 .5-.22.5-.5s-.22-.5-.5-.5-.5.22-.5.5.22.5.5.5z"
        />
        <circle
          cx="18"
          cy="10"
          r="1"
        />
        <circle
          cx="18"
          cy="6"
          r="1"
        />
        <path
          d="M21 10.5c.28 0 .5-.22.5-.5s-.22-.5-.5-.5-.5.22-.5.5.22.5.5.5z"
        />
        <circle
          cx="10"
          cy="6"
          r="1"
        />
        <circle
          cx="18"
          cy="14"
          r="1"
        />
        <circle
          cx="6"
          cy="18"
          r="1"
        />
        <path
          d="M14 20.5c-.28 0-.5.22-.5.5s.22.5.5.5.5-.22.5-.5-.22-.5-.5-.5zm7-7c-.28 0-.5.22-.5.5s.22.5.5.5.5-.22.5-.5-.22-.5-.5-.5zm-18 0c-.28 0-.5.22-.5.5s.22.5.5.5.5-.22.5-.5-.22-.5-.5-.5z"
        />
        <circle
          cx="10"
          cy="18"
          r="1"
        />
        <path
          d="M3 9.5c-.28 0-.5.22-.5.5s.22.5.5.5.5-.22.5-.5-.22-.5-.5-.5zm7 11c-.28 0-.5.22-.5.5s.22.5.5.5.5-.22.5-.5-.22-.5-.5-.5z"
        />
        <circle
          cx="6"
          cy="14"
          r="1"
        />
        <path
          d="M2.5 5.27L6 8.77l.28.28L6 9c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1c0-.1-.03-.19-.06-.28l2.81 2.81c-.71.11-1.25.73-1.25 1.47 0 .83.67 1.5 1.5 1.5.74 0 1.36-.54 1.47-1.25l2.81 2.81c-.09-.03-.18-.06-.28-.06-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1c0-.1-.03-.19-.06-.28l3.78 3.78h.01l1.41-1.41L3.91 3.86 2.5 5.27z"
        />
      </svg>
    </span>
  </button>
</div>
`;
